src/System.Private.CoreLib/src/System/Activator.cs

```diff
@@ -77,7 +78,26 @@ public static T CreateInstance<T>()
         }
 
         [Intrinsic]
-        private extern static T CreateInstanceIntrinsic<T>();
+        private static T CreateInstanceIntrinsic<T>()
+        {
+            // Fallback implementation for codegens that don't support this intrinsic.
+            // This uses the type loader and doesn't have the kind of guarantees about it always working
+            // as the intrinsic expansion has. Also, it's slower.
+
+            EETypePtr eetype = EETypePtr.EETypePtrOf<T>();
+
+            // The default(T) check can be evaluated statically and will result in the body of this method
+            // becoming empty for valuetype Ts. We still need a dynamic IsNullable check to cover Nullables though.
+            // This will obviously need work once we start supporting default valuetype constructors.
```

Comment:
> Do we have missing parts for the default valuetype constructors

I think it's all over the place. E.g. Array.Initialize won't run them, USG and the type loader might have implicit assumptions about them being not present, etc. On Project N, we fail the compilation if there's default valuetype constructor anywhere. If anyone looked into it, it would be @davidwrighton. I know there was a brief undertaking to look into it (that was quickly cut when C# cut the feature due to CoreCLR's bug around them).